aws/aws.go

```diff
@@ -766,23 +804,35 @@ func (s *awsOps) Attach(volumeID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	req := &ec2.AttachVolumeInput{
-		Device:     &devices[0],
-		InstanceId: &s.instance,
-		VolumeId:   &volumeID,
-	}
-	if _, err := s.ec2.AttachVolume(req); err != nil {
-		return "", err
-	}
-	vol, err := s.waitAttachmentStatus(
-		volumeID,
-		ec2.VolumeAttachmentStateAttached,
-		time.Minute,
-	)
-	if err != nil {
-		return "", err
+
+	for _, device := range devices {
+		req := &ec2.AttachVolumeInput{
+			Device:     &device,
+			InstanceId: &s.instance,
+			VolumeId:   &volumeID,
+		}
+		if _, err := s.ec2.AttachVolume(req); err != nil {
+			if strings.Contains(err.Error(), "is already in use") {
```

Comment:
nit: make sure this error string does not match with error returned when attach command fails because volume is attached on another node/already in use.

Reply:
Just tested that and we are good. If the volume is being used by another instance, it returns

```
ERRO[0000] failed to attach: VolumeInUse: vol-049cbf86e507578d2 is already attached to an instance
        status code: 400, request id: 0a85d578-0422-46a1-8f1c-50364a7679ba
```